src/netconfig.cpp

```diff
@@ -0,0 +1,401 @@
+                   domain);
+    }
+    else
+    {
+        bus.remove(object.c_str(), Dbus::ethInterface, Dbus::ethDomainName,
+                   domain);
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Set default gateway: `getway IP` */
+static void cmdGateway(Dbus& bus, Arguments& args)
+{
+    const auto [ver, ip] = args.asIpAddress();
+    args.expectEnd();
+
+    printf("Setting default gateway for IPv%i to %s...\n",
+           static_cast<int>(ver), ip);
+
+    const char* property =
+        ver == IpVer::v4 ? Dbus::syscfgDefGw4 : Dbus::syscfgDefGw6;
+
+    bus.set(Dbus::objectConfig, Dbus::syscfgInterface, property, ip);
+
+    puts(completeMessage);
+}
+
+/** @brief Add/remove IP: `ip [VLANID] {add|del} IP[/MASK GATEWAY]` */
+static void cmdIp(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Action action = args.asAction();
+
+    if (action == Action::add)
+    {
+        const auto [ipVer, ip, mask] = args.asIpAddrMask();
+        const auto [gwVer, gwIp] = args.asIpAddress();
+        args.expectEnd();
+        if (ipVer != gwVer)
+        {
+            throw std::invalid_argument("IP version mismatch");
+        }
+
+        const char* ipInterface =
+            ipVer == IpVer::v4 ? Dbus::ip4Interface : Dbus::ip6Interface;
+
+        bus.call(object.c_str(), Dbus::ipCreateInterface, Dbus::ipCreateMethod,
+                 ipInterface, ip, mask, gwIp);
+    }
+    else
+    {
+        const auto [_, ip] = args.asIpAddress();
+        args.expectEnd();
+
+        // Search for IP address' object
+        bool handled = false;
+        for (const auto& it : bus.getAddresses(object.c_str()))
+        {
+            if (it.address == ip)
+            {
+                bus.call(it.object.c_str(), Dbus::deleteInterface,
+                         Dbus::deleteMethod);
+                handled = true;
+                break;
+            }
+        }
+        if (!handled)
+        {
+            std::string err = "IP address ";
+            err += ip;
+            err += " not found";
+            throw std::invalid_argument(err);
+        }
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP client: 'dhcp [VLANID] {enable|disable}` */
+static void cmdDhcp(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Toggle toggle = args.asToggle();
+    args.expectEnd();
+
+    const bool enable = toggle == Toggle::enable;
+    printf("%s DHCP client...\n", enable ? "Enable" : "Disable");
+
+    bus.set(object.c_str(), Dbus::ethInterface, Dbus::ethDhcpEnabled, enable);
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP features: 'dhcpcfg {enable|disable} {dns|ntp}` */
+static void cmdDhcpcfg(Dbus& bus, Arguments& args)
+{
+    const char* featureDns = "dns";
+    const char* featureNtp = "ntp";
+
+    const std::string object = pathFromOptional(args);
```

Comment:
Seems, `Dbus::objectDhcp` should be here.

Reply:
Yeah, copy-paste. Thanks!